### PR TITLE
GCP Auth: Modifying incorrect audience string for GCE

### PR DESCRIPTION
**Pull request by conor-mccullough**
### Description
Initially raised by ent customer via [issue 2055](https://github.com/hashicorp/terraform-provider-vault/issues/2055), then via Zendesk in an official support engagement.

Unsure whether this requires a changelog entry - please let me know if so and I'll update the PR.

### Output from acceptance testing:

Test results uploaded internally via Jira, can upload here too if needed

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Review by fairclothjm — COMMENTED** [review]
Thanks @conor-mccullough! Yes, we would appreciate a changelog entry!

**Comment by conor-mccullough** [comment]
Cheers @fairclothjm - Added to the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ BUGS:
 * Fix panic when readnig client_secret from a public oidc client ([#2048](https://github.com/hashicorp/terraform-provider-vault/pull/2048))
 * Fix API request missing `roles` field for `mongodbatlas_secret_role` resource ([#2047](https://github.com/hashicorp/terraform-provider-vault/pull/2047))
 * Fix bug when updating vault_azure_secret_backend_role: ([#2063](https://github.com/hashicorp/terraform-provider-vault/pull/2063))
+* Fix string ordering for `gcp_auth` causing GCE auth to fail ([#2064](https://github.com/hashicorp/terraform-provider-vault/pull/2064))
```
Comment (fairclothjm) [inline CHANGELOG.md:11]:
```suggestion
* Fix audience string ordering for `auth_login_gcp` causing GCE auth to fail ([#2064](https://github.com/hashicorp/terraform-provider-vault/pull/2064))
```

**Comment by OneWhoNests** [comment]
fixes #2055

**Pull request merged**